google-cloud-bigquery/lib/google/cloud/bigquery/external.rb

```diff
@@ -0,0 +1,2200 @@
+          #
+          # @example
+          #   require "google/cloud/bigquery"
+          #
+          #   bigquery = Google::Cloud::Bigquery.new
+          #
+          #   csv_url = "gs://bucket/path/to/data.csv"
+          #   csv_table = bigquery.external csv_url do |csv|
+          #     csv.ignore_unknown = true
+          #   end
+          #
+          #   csv_table.ignore_unknown #=> true
+          #
+          def ignore_unknown= new_ignore_unknown
+            @gapi.ignore_unknown_values = new_ignore_unknown
+          end
+
+          ##
+          # The maximum number of bad records that BigQuery can ignore when
+          # reading data. If the number of bad records exceeds this value, an
+          # invalid error is returned in the job result. The default value is 0,
+          # which requires that all records are valid. This setting is ignored
+          # for Google Cloud Bigtable, Google Cloud Datastore backups and Avro
+          # formats.
+          #
+          # @return [Integer]
+          #
+          # @example
+          #   require "google/cloud/bigquery"
+          #
+          #   bigquery = Google::Cloud::Bigquery.new
+          #
+          #   csv_url = "gs://bucket/path/to/data.csv"
+          #   csv_table = bigquery.external csv_url do |csv|
+          #     csv.max_bad_records = 10
+          #   end
+          #
+          #   csv_table.max_bad_records #=> 10
+          #
+          def max_bad_records
+            @gapi.max_bad_records
+          end
+
+          ##
+          # Set the maximum number of bad records that BigQuery can ignore when
+          # reading data. If the number of bad records exceeds this value, an
+          # invalid error is returned in the job result. The default value is 0,
+          # which requires that all records are valid. This setting is ignored
+          # for Google Cloud Bigtable, Google Cloud Datastore backups and Avro
+          # formats.
+          #
+          # @param [Integer] new_max_bad_records New max_bad_records value
+          #
+          # @example
+          #   require "google/cloud/bigquery"
+          #
+          #   bigquery = Google::Cloud::Bigquery.new
+          #
+          #   csv_url = "gs://bucket/path/to/data.csv"
+          #   csv_table = bigquery.external csv_url do |csv|
+          #     csv.max_bad_records = 10
+          #   end
+          #
+          #   csv_table.max_bad_records #=> 10
+          #
+          def max_bad_records= new_max_bad_records
+            @gapi.max_bad_records = new_max_bad_records
+          end
+
+          ##
+          # @private Google API Client object.
+          def to_gapi
+            @gapi
+          end
+        end
+
+        ##
+        # # CsvTable
+        #
+        # {External::CsvTable} is a subclass of {External::Table} and represents
+        # a CSV external data source that can be queried from directly, even
```

Comment:
"external data source" is vague, so it might be good to state that the file must be in Google Cloud Storage or Google Drive.

Reply:
Added.